### PR TITLE
Add static framework targets to all modules

### PR DESCRIPTION
**Pull request by russelarms**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add 3 OS-specific targets to each module (21 total).
For now we only add new static framework targets for iOS, tvOS, macOS, to each module.
Note that the changes should be contained in .xconfig files, otherwise build would fail.
Those targets should contain static librarty target and public headers. Another option is to add sources instead of static library targets - but that would increase complexity of maintaining.

## Related PRs or issues

[AB#66952](https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/66952)

### CODE REVIEW

File: AppCenterCrashes/AppCenterCrashesStaticIOS copy-Info.plist [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashesStaticIOS copy-Info.plist]: ` copy-Info.plist` 😄

File: Scripts/AppCenter/build-appcenter-ios-framework.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-source $(dirname "$0")/../build-ios-framework.sh AppCenter
+source $(dirname "$0")/../build-ios-framework.sh AppCenterFramework
```
Comment (MatkovIvan) [inline Scripts/AppCenter/build-appcenter-ios-framework.sh]: `AppCenterFramework.framework`? Are you sure?

File: Config/Static Framework.xcconfig [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CURRENT_PROJECT_VERSION = 1
+DEFINES_MODULE = YES
+DYLIB_COMPATIBILITY_VERSION = 1
+DYLIB_CURRENT_VERSION = 1
```
Comment (MatkovIvan) [inline Config/Static Framework.xcconfig]: Should we use our version instead?

Comment (clpolet) [reply Config/Static Framework.xcconfig]: Good point, please use our version.

**Pull request closed without merging**